analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DisablingRequestValidationBase.cs

```diff
@@ -0,0 +1,161 @@
+                        && a.ConstructorArguments[0].Value is bool b
+                        && b == false
+                        && a.AttributeClass.Is(KnownType.System_Web_Mvc_ValidateInputAttribute));
+                    if (attributeWithFalseParameter != null)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attributeWithFalseParameter.ApplicationSyntaxReference.GetSyntax().GetLocation()));
+                    }
+                },
+                SymbolKind.NamedType,
+                SymbolKind.Method);
+
+            context.RegisterCompilationAction(c =>
+            {
+                if (!IsEnabled(c.Options))
+                {
+                    return;
+                }
+
+                foreach (var fullPath in GetWebConfigFilePathsRecursively(rootPath))
+                {
+                    var webConfig = File.ReadAllText(fullPath);
+                    if (webConfig.Contains("<system.web>") && ParseXDocument(webConfig) is { } doc)
+                    {
+                        ReportValidateRequest(doc, fullPath, c);
+                        ReportRequestValidationMode(doc, fullPath, c);
+                    }
+                }
+            });
+        }
+
+        private void ReportValidateRequest(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var pages in doc.XPathSelectElements("configuration/system.web/pages"))
+            {
+                if (pages.Attribute("validateRequest") is { } validateRequest
+                    && validateRequest.Value.Equals("false", StringComparison.OrdinalIgnoreCase)
+                    && CreateLocation(webConfigPath, pages, validateRequest.ToString()) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+        }
+
+        private void ReportRequestValidationMode(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var httpRuntime in doc.XPathSelectElements("configuration/system.web/httpRuntime"))
+            {
+                if (httpRuntime.Attribute("requestValidationMode") is { } requestValidationMode
+                    && decimal.TryParse(requestValidationMode.Value, out var value)
+                    && value < MinimumAcceptedRequestValidationModeValue
+                    && CreateLocation(webConfigPath, httpRuntime, requestValidationMode.ToString()) is { } location)
+                {
+                    c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetWebConfigFilePathsRecursively(string rootPath) =>
+            Directory.GetFiles(rootPath, "web.config", SearchOption.AllDirectories).Select(p => Path.GetFullPath(p));
```

Comment:
`GetFullPath` should happen for the root in the constructor once like in the sonar security. Like this, it's evaluated as many times as we can find web.config files.
```suggestion
            Directory.GetFiles(rootPath, "web.config", SearchOption.AllDirectories);
```